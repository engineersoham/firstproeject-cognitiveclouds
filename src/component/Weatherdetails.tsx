import React, { useEffect, useState } from 'react'
import { Box } from '@mui/material';
import { useParams } from 'react-router-dom';
import { Typography, Avatar, Card, CardContent } from '@mui/material';



const Weatherdetails = () => {

    const { capital } = useParams()

    const [weatherdata, setweather] = useState(
        {
            capital: '',
            Temp: '',
            Whether_icon: '',
            Wind_speed: '',
            precip: 0
        }
    )

    const getweather = (capital: string) => {
        fetch(`http://api.weatherstack.com/current?access_key=d81381e7f65ac20a3115570356e0a315&query=${capital}`)
            .then(res => res.json())
            .then((data) => {
                setweather(
                    {
                        ...weatherdata,
                        capital: data?.location?.name,
                        Temp: data?.current?.temperature,
                        Whether_icon: data?.current?.weather_icons[0],
                        Wind_speed: data?.current.wind_speed,
                        precip: data?.current.precip
                    }
                )
                
            })
    }

    useEffect(() => {
        if (capital) {
            getweather(capital)
        }

    }, [capital])

    return (
        <div>
            <h3 style={{ textAlign: 'center', marginBottom: '50px' }}> Current Weather Report of {weatherdata.capital}</h3>


            <Card style={{ boxShadow: '5px 10px #888888' }} variant='outlined' sx={{ display: 'flex', ml: 70, mr: 70 }}>
                <div style={{ float: 'left' }}>
                    <Box sx={{ display: 'flex', flexDirection: 'column' }}>
                        <CardContent sx={{ flex: '1 0 auto' }}>
                            <Typography component="div" variant="h6">
                                City: {weatherdata.capital}.
                            </Typography>
                            <Typography variant="h6" color="text.secondary" component="div">
                                Temparature: {weatherdata.Temp} Degree Celsius
                            </Typography>
                            <Typography variant="h6" color="text.secondary" component="div">
                                Wind Speed: {weatherdata.Wind_speed} k/h
                            </Typography>
                            <Typography variant="h6" color="text.secondary" component="div">
                                Precip: {weatherdata.precip}
                            </Typography>
                        </CardContent>

                    </Box>
                </div>
                <div style={{ float: 'right' }}>
                    <Avatar
                        variant="rounded"
                        src={weatherdata.Whether_icon}
                        sx={{ width: 56, height: 56 ,mt:5}}
                    />
                </div>


            </Card>

        </div>
    )
}

export default Weatherdetails