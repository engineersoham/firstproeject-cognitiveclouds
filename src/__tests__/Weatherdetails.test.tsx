import {screen, render} from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Weatherdetails from '../component/Weatherdetails'

test('snapshot for weathercomponent', ()=>{
    const weathersnap = render(
        <BrowserRouter>
        <Weatherdetails/>
        </BrowserRouter>
    );

    expect(weathersnap).toMatchSnapshot()
})

describe('test for weather component', () => {
    it('renderring weather', () => {
      render(
        <BrowserRouter>
           <Weatherdetails/>
        </BrowserRouter>
      );
      screen.debug();
    })
  })