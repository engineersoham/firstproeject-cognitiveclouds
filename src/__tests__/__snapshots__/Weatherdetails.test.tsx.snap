// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for weathercomponent 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3
          style="text-align: center; margin-bottom: 50px;"
        >
           Current Weather Report of 
        </h3>
        <div
          class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1iauk4v-MuiPaper-root-MuiCard-root"
          style="box-shadow: 5px 10px #888888;"
        >
          <div
            style="float: left;"
          >
            <div
              class="MuiBox-root css-j7qwjs"
            >
              <div
                class="MuiCardContent-root css-1qaznj3-MuiCardContent-root"
              >
                <div
                  class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                >
                  City: 
                  .
                </div>
                <div
                  class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
                >
                  Temparature: 
                   Degree Celsius
                </div>
                <div
                  class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
                >
                  Wind Speed: 
                   k/h
                </div>
                <div
                  class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
                >
                  Precip: 
                  0
                </div>
              </div>
            </div>
          </div>
          <div
            style="float: right;"
          >
            <div
              class="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault css-1r24hxp-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3
        style="text-align: center; margin-bottom: 50px;"
      >
         Current Weather Report of 
      </h3>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1iauk4v-MuiPaper-root-MuiCard-root"
        style="box-shadow: 5px 10px #888888;"
      >
        <div
          style="float: left;"
        >
          <div
            class="MuiBox-root css-j7qwjs"
          >
            <div
              class="MuiCardContent-root css-1qaznj3-MuiCardContent-root"
            >
              <div
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                City: 
                .
              </div>
              <div
                class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
              >
                Temparature: 
                 Degree Celsius
              </div>
              <div
                class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
              >
                Wind Speed: 
                 k/h
              </div>
              <div
                class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
              >
                Precip: 
                0
              </div>
            </div>
          </div>
        </div>
        <div
          style="float: right;"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault css-1r24hxp-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;